demo/DemoArticles/update.js

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env node
+/* eslint-disable require-jsdoc */
+
+const fs = require('fs')
+const request = require('request')
+const indexJSON = require('./articles.json')
+const ArticleRef = require('./ArticleRef.js').ArticleRef
+
+const articleRefs = indexJSON
+  .map(articleData => new ArticleRef(articleData.lang, articleData.title, articleData.revision))
+
+// eslint-disable-next-line no-console
+console.log(`Fetching JSON for ${articleRefs.length} titles...`)
+
+// Linting complained of some improperly formatted characters which turned out to be these.
+const escapeLangDirectionMarks = string => string
+  .replace(/\u00ad/g, '\\u00ad')
+  .replace(/\u200c/g, '\\u200c')
+  .replace(/\u200e/g, '\\u200e')
+  .replace(/\u200f/g, '\\u200f')
+  .replace(/\u202d/g, '\\u202d')
+  .replace(/\u202e/g, '\\u202e')
+  .replace(/\u202f/g, '\\u202f')
+
+const saveJSONForArticleRef = (articleJSON, articleRef) => {
+  const formattedArticleJSON = JSON.stringify(articleJSON, null, 2)
+  const fullyEscapedArticleJSON = escapeLangDirectionMarks(formattedArticleJSON)
+  fs.writeFile(
+    `./data/${articleRef.fileName()}`,
+    fullyEscapedArticleJSON, err => { if (err) { throw err } }
+  )
+  // eslint-disable-next-line no-console
+  console.log(`\tJSON saved to '${articleRef.fileName()}'`)
+}
+
+const fetchAndSaveJSONForArticleRef = articleRef => {
+  const requestOptions = {
+    method: 'POST',
+    uri: articleRef.url(),
+    encoding: 'utf-8',
+    gzip: true
+  }
+  const responseHandler = (error, response, body) => {
+    if (!error && response.statusCode === 200) {
```

Comment:
Consider throwing on error or unexpected response to be consistent with error handling of `fs.writeFile()`.